frontend/src/app/shared/components/fields/edit/field-types/hours-duration-edit-field.component.ts

```diff
@@ -63,20 +66,27 @@ export class HoursDurationEditFieldComponent extends EditFieldComponent {
     // context, we check the validity of the input and, if it's not valid, we
     // default to the previous value, emulating the way the browsers work with
     // valid separators (e.g: introducing 1. would set 1 as a value).
-    if (value == null && !input.validity.valid) {
-      value = this.value || 0;
+    this.inputValue = input.value;
+    if (!input.validity.valid) {
+      if (value === null || input.value === '') {
+        value = null;
+      } else {
+        value = this.value as string;
+      }
     }
-
     return moment.duration(value, 'hours');
   }
 
   public formatter(value:any) {
```

Comment:
Perhaps the typing can be more explicit here, so that the check below is easier to understand